pkg/distributor/otel.go

```diff
@@ -72,10 +73,12 @@ func OTLPHandler(
 		switch contentEncoding {
 		case "gzip":
 			compression = util.Gzip
+		case "lz4":
+			compression = util.Lz4
 		case "":
 			compression = util.NoCompression
 		default:
-			return httpgrpc.Errorf(http.StatusUnsupportedMediaType, "unsupported compression: %s. Only \"gzip\" or no compression supported", contentEncoding)
+			return httpgrpc.Errorf(http.StatusUnsupportedMediaType, "unsupported compression: %s. \"gzip\", \"lz4\", or no compression supported", contentEncoding)
```

Comment:
The failing test made me realize I don't want to drop "Only" from the error message:
```suggestion
			return httpgrpc.Errorf(http.StatusUnsupportedMediaType, "unsupported compression: %s. Only \"gzip\", \"lz4\", or no compression supported", contentEncoding)
```